controllers/instanaagent_controller.go

```diff
@@ -57,18 +70,23 @@ func Add(mgr manager.Manager) error {
 func add(mgr ctrl.Manager, r *InstanaAgentReconciler) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&instanaV1.InstanaAgent{}).
-		Owns(&appV1.DaemonSet{}, builder.WithPredicates(filterPredicate())).
+		Owns(&appV1.DaemonSet{}).
 		Owns(&coreV1.Pod{}).
-		Owns(&coreV1.Secret{}).
 		Owns(&coreV1.ConfigMap{}).
-		Owns(&coreV1.Service{}).
+		Owns(&coreV1.Secret{}).
 		Owns(&coreV1.ServiceAccount{}).
+		Owns(&coreV1.Service{}).
+		Owns(&v1.ClusterRole{}).
+		Owns(&v1.ClusterRoleBinding{}).
+		WithEventFilter(filterPredicate()).
```

Comment:
Fixed the event filtering. Prevents the reconcile loop getting called just for status-field updates.